src/lib/utils/time.ts

```diff
@@ -1,18 +1,17 @@
+import { LanguageTypeEnum } from 'constants/LanguageTypeEnum'
 import dayjs from 'dayjs'
-import durationPlugin from 'dayjs/plugin/duration'
+import duration from 'dayjs/plugin/duration'
+import relativeTime from 'dayjs/plugin/relativeTime'
+import { selectSelectedLocale } from 'state/slices/selectors'
+import { store } from 'state/store'
+
+dayjs.extend(duration)
+dayjs.extend(relativeTime)
 
 export const formatSecondsToDuration = (seconds: number) => {
-  dayjs.extend(durationPlugin)
-  const duration = dayjs.duration(seconds, 'seconds')
-  const hours = duration.asHours()
-  const days = duration.asDays()
-  const months = duration.asMonths()
+  const selectedLocale = selectSelectedLocale(store.getState())
+  const locale = selectedLocale in LanguageTypeEnum ? selectedLocale : 'en'
+  require(`dayjs/locale/${locale}.js`)
 
-  if (hours < 24) {
-    return `${Math.floor(hours)} hour${Math.floor(hours) !== 1 ? 's' : ''}`
-  } else if (days < 31) {
-    return `${Math.floor(days)} day${Math.floor(days) !== 1 ? 's' : ''}`
-  } else {
-    return `${Math.floor(months)} month${Math.floor(months) !== 1 ? 's' : ''}`
-  }
+  return dayjs.duration(seconds, 'seconds').locale(locale).humanize()
```

Comment:
If I remember correctly you don't need to load the locale and set it again, it's set and loaded in the AppContext and updated if the user changes the language... but it shouldn't hurt:

https://github.com/shapeshift/web/blob/11b84c96a60cdcd0b9c8801c8fd327c19c07f466/src/context/AppProvider/AppContext.tsx#L85-L90

Reply:
Yeah we could assume things are loaded here which they *should* but this is a pure util i.e it shouldn't know about its parent context (plus this require is cheap and a no-op if we already required that module and have it cached)